### ISSUE TITLE
Removing a string value (changing it to nothing) leads to an error

### ISSUE THREAD

**Issue by ocrasborn** [Woseseltops/signbank#25]
A rather nasty error, in fact, on my machine: I can't click away the error message, have to hard-quit my browser and restart it.
![screenshot 2015-01-03 22 45 00](https://cloud.githubusercontent.com/assets/10242207/5603899/9347a6b0-939a-11e4-9bfe-e55e28f2afbd.png)

**Comment by Woseseltops** [comment]
Good find, a stupid mistake on my part. I don't know why you could not close the browser window, though, that must have been a Safari thingy.

**Comment by ocrasborn** [comment]
I now get the message below for _every_ change I try to make, whether in a text field or in selecting a(nother) value from a list. Both in Firefox and Safari, but in FF I can click away the error box.

-----------------
There was an error processing this change: UnicodeEncodeError at /dictionary/update/gloss/2
'ascii' codec can't encode character u'\xc9' in position 10: ordinal not in range(128)

Request Method: POST
Request URL: http://signbank.science.ru.nl/dictionary/update/gloss/2
Django Version: 1.6.1
Python Executable: /usr/bin/python
Python Version: 2.7.3
Python Path: ['/var/www2/signbank/live/virtualenv/signbank/lib/python2.7/site-packages/distribute-0.6.24-py2.7.egg', '/var/www2/signbank/live/virtualenv/signbank/lib/python2.7/site-packages/pip-1.1-py2.7.egg', '/var/www2/signbank/live/virtualenv/signbank/lib/python2.7/site-packages', '/usr/lib/python2.7', '/usr/lib/python2.7/plat-linux2', '/usr/lib/python2.7/lib-tk', '/usr/lib/python2.7/lib-old', '/usr/lib/python2.7/lib-dynload', '/usr/local/lib/python2.7/dist-packages', '/usr/lib/python2.7/dist-packages', '/usr/lib/python2.7/dist-packages/PIL', '/usr/lib/python2.7/dist-packages/gtk-2.0', '/usr/lib/pymodules/python2.7']
Server time: Tue, 20 Jan 2015 01:37:40 +1100
Installed Applications:
('django.contrib.auth',
 'django.contrib.contenttypes',
 'django.contrib.sessions',
 'django.contrib.sites',
 'django.contrib.messages',
 'django.contrib.admin',
 'django.contrib.admindocs',
 'django.contrib.staticfiles',
 'bootstrap3',
 'django_summernote',
 'signbank.dictionary',
 'signbank.feedback',
 'signbank.registration',
 'signbank.pages',
 'signbank.attachments',
 'signbank.video',
 'south',
 'reversion',
 'django_mobile',
 'tagging')
Installed Middleware:
('django.middleware.common.CommonMiddleware',
 'django.contrib.sessions.middleware.SessionMiddleware',
 'django.middleware.csrf.CsrfViewMiddleware',
 'django.contrib.auth.middleware.AuthenticationMiddleware',
 'django.contrib.messages.middleware.MessageMiddleware',
 'django.middleware.clickjacking.XFrameOptionsMiddleware',
 'signbank.pages.middleware.PageFallbackMiddleware',
 'django_mobile.middleware.MobileDetectionMiddleware',
 'django_mobile.middleware.SetFlavourMiddleware',
 'reversion.middleware.RevisionMiddleware')

Traceback:
File "/var/www2/signbank/live/virtualenv/signbank/lib/python2.7/site-packages/django/core/handlers/base.py" in get_response
  201.                 response = middleware_method(request, response)
File "/var/www2/signbank/live/virtualenv/signbank/lib/python2.7/site-packages/reversion/middleware.py" in process_response
  37.         self._close_revision(request)
File "/var/www2/signbank/live/virtualenv/signbank/lib/python2.7/site-packages/reversion/middleware.py" in _close_revision
  28.             revision_context_manager.end()
File "/var/www2/signbank/live/virtualenv/signbank/lib/python2.7/site-packages/reversion/revisions.py" in end
  182.                             db = self._db,
File "/var/www2/signbank/live/virtualenv/signbank/lib/python2.7/site-packages/reversion/revisions.py" in save_revision
  455.                     objects[obj] = adapter.get_version_data(obj)
File "/var/www2/signbank/live/virtualenv/signbank/lib/python2.7/site-packages/reversion/revisions.py" in get_version_data
  109.             "object_repr": force_text(obj),
File "/var/www2/signbank/live/virtualenv/signbank/lib/python2.7/site-packages/django/utils/encoding.py" in force_text
  108.                     s = six.text_type(bytes(s), encoding, errors)
File "/var/www2/signbank/live/virtualenv/signbank/lib/python2.7/site-packages/signbank/dictionary/models.py" in __str__
  32.         return str(self.gloss)+"-"+str(self.translation)

Exception Type: UnicodeEncodeError at /dictionary/update/gloss/2
Exception Value: 'ascii' codec can't encode character u'\xc9' in position 10: ordinal not in range(128)
Request information:
GET: No GET data

POST:
csrfmiddlewaretoken = u'QqpdALOGmq83SQhAl04MvfC8sQCl5bV1'
id = u'namEnt'
value = u'_3'

FILES: No FILES data

COOKIES:
csrftoken = 'QqpdALOGmq83SQhAl04MvfC8sQCl5bV1'
__utma = '214840340.1044250266.1393406043.1414573345.1415088329.66'
__utmz = '214840340.1401693407.45.9.utmcsr=google|utmccn=(organic)|utmcmd=organic|utmctr=(not%20provided)'
_ga = 'GA1.2.1044250266.1393406043'
sessionid = 'wbnnizi0x78f1ct95y3r3alvb1wjdmxj'

META:
mod_wsgi.listener_port = '80'
HTTP_REFERER = 'http://signbank.science.ru.nl/dictionary/gloss/2/'
mod_wsgi.listener_host = '131.174.16.34'
mod_wsgi.handler_script = ''
SERVER_SOFTWARE = 'Apache/2.2.22 (Ubuntu)'
SCRIPT_NAME = u''
HTTP_X_CSRFTOKEN = 'QqpdALOGmq83SQhAl04MvfC8sQCl5bV1'
SERVER_SIGNATURE = '<address>Apache/2.2.22 (Ubuntu) Server at signbank.science.ru.nl Port 80</address>\n'
REQUEST_METHOD = 'POST'
PATH_INFO = u'/dictionary/update/gloss/2'
SERVER_PROTOCOL = 'HTTP/1.1'
QUERY_STRING = ''
CONTENT_LENGTH = '71'
HTTP_USER_AGENT = 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10.10; rv:34.0) Gecko/20100101 Firefox/34.0'
HTTP_CONNECTION = 'keep-alive'
HTTP_COOKIE = '__utma=214840340.1044250266.1393406043.1414573345.1415088329.66; __utmz=214840340.1401693407.45.9.utmcsr=google|utmccn=(organic)|utmcmd=organic|utmctr=(not%20provided); _ga=GA1.2.1044250266.1393406043; csrftoken=QqpdALOGmq83SQhAl04MvfC8sQCl5bV1; sessionid=wbnnizi0x78f1ct95y3r3alvb1wjdmxj'
SERVER_NAME = 'signbank.science.ru.nl'
REMOTE_ADDR = '131.174.132.234'
mod_wsgi.request_handler = 'wsgi-script'
wsgi.url_scheme = 'http'
PATH_TRANSLATED = '/var/www2/signbank/live/repo/signbank/signbank/wsgi.py/dictionary/update/gloss/2'
SERVER_PORT = '80'
wsgi.multiprocess = False
mod_wsgi.input_chunked = '0'
SERVER_ADDR = '131.174.16.34'
DOCUMENT_ROOT = '/var/www2/signbank/live/htdocs'
mod_wsgi.process_group = 'signbank'
HTTP_PRAGMA = 'no-cache'
HTTP_X_REQUESTED_WITH = 'XMLHttpRequest'
SCRIPT_FILENAME = '/var/www2/signbank/live/repo/signbank/signbank/wsgi.py'
SERVER_ADMIN = '[no address given]'
wsgi.input = <mod_wsgi.Input object at 0x7f9c6caf3530>
HTTP_DNT = '1'
HTTP_HOST = 'signbank.science.ru.nl'
wsgi.multithread = True
mod_wsgi.callable_object = 'application'
HTTP_CACHE_CONTROL = 'no-cache'
REQUEST_URI = '/dictionary/update/gloss/2'
HTTP_ACCEPT = 'text/html, */*; q=0.01'
wsgi.version = 
GATEWAY_INTERFACE = 'CGI/1.1'
wsgi.run_once = False
wsgi.errors = <mod_wsgi.Log object at 0x7f9c65519b30>
REMOTE_PORT = '58662'
HTTP_ACCEPT_LANGUAGE = 'nl,en-US;q=0.7,en;q=0.3'
mod_wsgi.version = 
CONTENT_TYPE = 'application/x-www-form-urlencoded; charset=UTF-8'
mod_wsgi.application_group = 'signbank.science.ru.nl|'
mod_wsgi.script_reloading = '1'
wsgi.file_wrapper = ''
CSRF_COOKIE = u'QqpdALOGmq83SQhAl04MvfC8sQCl5bV1'
HTTP_ACCEPT_ENCODING = 'gzip, deflate'

Settings:
Using settings module signbank.settings.development
FORCE_LOWERCASE_TAGS = True
USE_L10N = True
ADMIN_RESULT_FIELDS = ['sn', 'idgloss', 'annotation_idgloss', 'morph']
TIME_ZONE = 'Australia/Sydney'
CSRF_COOKIE_SECURE = False
LANGUAGE_CODE = 'en-us'
ROOT_URLCONF = 'signbank.urls'
MANAGERS = ('Wessel Stoop', 'w.stoop@let.ru.nl')
BASE_DIR = '/var/www2/signbank/live/virtualenv/signbank/lib/python2.7/site-packages/signbank'
DEFAULT_CHARSET = 'utf-8'
SESSION_SERIALIZER = 'django.contrib.sessions.serializers.JSONSerializer'
STATIC_ROOT = ''
FFMPEG_TIMEOUT = 60
LOG_FILENAME = 'debug.log'
USE_THOUSAND_SEPARATOR = False
ALLOWED_HOSTS = []
MESSAGE_STORAGE = 'django.contrib.messages.storage.fallback.FallbackStorage'
EMAIL_SUBJECT_PREFIX = '[Django] '
SEND_BROKEN_LINK_EMAILS = False
STATICFILES_FINDERS = 
SESSION_CACHE_ALIAS = 'default'
ATTACHMENT_LOCATION = 'attachments'
SESSION_COOKIE_DOMAIN = None
SESSION_COOKIE_NAME = 'sessionid'
CSRF_COOKIE_NAME = 'csrftoken'
ADMIN_FOR = 
TIME_INPUT_FORMATS = 
AUSLAN_STATIC_PREFIX = '/static/'
DATABASES = {'default': {'ENGINE': 'django.db.backends.sqlite3', 'AUTOCOMMIT': True, 'ATOMIC_REQUESTS': False, 'NAME': '/var/www2/signbank/live/writable/database/signbank.db', 'TEST_MIRROR': None, 'OPTIONS': {}, 'CONN_MAX_AGE': 0, 'TIME_ZONE': 'UTC', 'TEST_COLLATION': None, 'TEST_CHARSET': None, 'HOST': '', 'USER': '', 'TEST_NAME': None, 'PASSWORD': u'********************', 'PORT': ''}}
FFMPEG_OPTIONS = ['-vcodec', 'h264', '-an']
FILE_UPLOAD_PERMISSIONS = None
FILE_UPLOAD_HANDLERS = 
DEFAULT_CONTENT_TYPE = 'text/html'
APPEND_SLASH = True
FIRST_DAY_OF_WEEK = 0
DATABASE_ROUTERS = []
SHOW_NUMBERSIGNS = True
FFMPEG_PROGRAM = '/Applications/ffmpegX.app/Contents/Resources/ffmpeg'
YEAR_MONTH_FORMAT = 'F Y'
STATICFILES_STORAGE = 'django.contrib.staticfiles.storage.StaticFilesStorage'
CACHES = {'default': {'BACKEND': 'django.core.cache.backends.locmem.LocMemCache'}}
PRIMARY_CSS = 'bootstrap_css/ngt.css'
SERVER_EMAIL = 'root@localhost'
SESSION_COOKIE_PATH = '/'
PAGES_VIDEO_LOCATION = 'pages'
MIDDLEWARE_CLASSES = 
USE_I18N = True
THOUSAND_SEPARATOR = ','
SECRET_KEY = u'********************'
XALLOWED_TAGS = ['', 'workflow:needs video', 'workflow:redo video', 'workflow:problematic', 'corpus:attested', 'lexis:doubtlex', 'phonology:alternating', 'phonology:dominant hand only', 'phonology:double handed', 'phonology:forearm rotation', 'phonology:handshape change', 'phonology:onehand', 'phonology:parallel', 'phonology:symmetrical', 'phonology:two handed']
LANGUAGE_COOKIE_NAME = 'django_language'
DEFAULT_INDEX_TABLESPACE = ''
TRANSACTIONS_MANAGED = False
LOGGING_CONFIG = 'django.utils.log.dictConfig'
SIGNING_BACKEND = 'django.core.signing.TimestampSigner'
TEMPLATE_LOADERS = 
WSGI_APPLICATION = 'signbank.wsgi.application'
TEMPLATE_DEBUG = True
X_FRAME_OPTIONS = 'SAMEORIGIN'
DO_LOGGING = False
FORCE_SCRIPT_NAME = None
USE_X_FORWARDED_HOST = False
ACCOUNT_ACTIVATION_DAYS = 7
MEDIA_MOBILE_URL = '/media/'
SESSION_COOKIE_SECURE = False
CSRF_COOKIE_DOMAIN = None
FILE_CHARSET = 'utf-8'
DEBUG = True
COMMENT_VIDEO_LOCATION = 'comments'
SESSION_FILE_PATH = None
DEFAULT_FILE_STORAGE = 'django.core.files.storage.FileSystemStorage'
INSTALLED_APPS = 
LANGUAGES_BIDI = 
VIDEO_UPLOAD_LOCATION = 'upload'
COMMENTS_ALLOW_PROFANITIES = False
DATABASE_NAME = '/www2/signbank/live/repo/signbank/signbank.db'
PREPEND_WWW = False
SECURE_PROXY_SSL_HEADER = None
SESSION_COOKIE_HTTPONLY = True
DEBUG_PROPAGATE_EXCEPTIONS = False
MONTH_DAY_FORMAT = 'F j'
LOGIN_URL = '/accounts/login/'
SESSION_EXPIRE_

**Comment by richardbank** [comment]
I only once got an error while editing a test-entry, but I can't reproduce it. 
Your error, Onno, has likely to do with character encoding (Request URL: http://signbank.science.ru.nl/dictionary/update/gloss/2 contains keyword 'Abbé de l'Épée' with unicode chars). This is related to #26 and #5 (the latter still has problems, see my comment there)

**Comment by Woseseltops** [comment]
@richardbank is correct. This problem is fixed, the problem @ocrasborn mentions is indeed a separate problem related to char encoding. It's fixed now!

**Comment by ocrasborn** [comment]
Same error popped up again. I enter a special character in the Keywords field (now renamed Translation Equivalents in the detail view), click OK, and then the same/similar message appears, blocking Safari. Interesting: upon restarting Safari, the word with the special character is correctly saved.

**Comment by ocrasborn** [comment]
Latest problem: can't enter anything in the Remarks field, it will always generate a safari-blocking error message, independent of whether there's special characters or non-alphabetic characters or not

**Comment by ocrasborn** [comment]
I have removed the Remarks field from the Detail view; yet to remove from search page. Remarks should be in a note, not in a separate field. At the same time, I removed two further references to SN / Sign
Number.

**Comment by Woseseltops** [comment]
Is there still something to here for me? If yes, maybe we can start a separate issue for it, because the problem doesn't seem to be anymore that 'removing a string value leads to an error'.
